### ISSUE TITLE
Question about using `early_obj_ptr` for `decoder_out_no_mem_cme` in CME

### ISSUE THREAD

**Issue by GinnyXiao** [ClaudiaCuttano/SAMWISE#30]
Dear authors,

Thanks for the wonderful work and for releasing the code! SAMWISE has deeply inspired me.

I wonder why in your CME implementation, you decided to use `decoder_out_no_mem_cme.early_obj_ptr` embedding to predict the logits? Why not using `decoder_out_no_mem_cme.obj_ptr` as you did for the memory path decoder output `decoder_out_w_mem`? If I understand correctly, `early_obj_ptr` = the first-layer mask token embedding (single-mask token) from `mask_decoder`; `obj_ptr` = the final mask token after the full transformer, projected by `obj_ptr_proj`.

**Comment by GinnyXiao** [comment]
Also in your paper in the sec 3.5 CME you wrote you "initialize (and keep frozen) the weights of the cross-attention with those from SAM2 Mask Decoder" for the CME module, but I didn't find your implementation regarding this. 

In samwise.py lines 471-474
```
# freeze all the weights except CMT adapter and Conditional Memory Encoder
    for param_name, param in model.named_parameters():
        if 'adapter' not in param_name and 'conditional_memory_encoder' not in param_name and 'project_text' not in param_name:
            param.requires_grad = False
```
it seems like CME is trained from scratch.

Thank you for your time and help!

**Comment by ga1i13o** [comment]
Dear Ginny,
Thanks for your kind words!
Your understanding is correct! 
`early_obj_ptr = the first-layer mask token embedding (single-mask token) from mask_decoder; obj_ptr = the final mask token after the full transformer, projected by obj_ptr_proj`

Is exactly what happens. We made that choice during development initially as in principle one layer of cross-attention should be enough to obtain a token that represents the memory-less features and we wanted to avoid the full forward pass in the decoder. Performance ( in terms of correctly minimizing the self-supervised loss) were good so we kept it that way.

What we meant with 'initialized and keep frozen' the weights ot the CA is that we simply re-used the CA from the Mask Decoder. We then train the CME that takes as input the two obj_pointers and a decision token.


Sorry for the delay, and let me know if you have further question
